### ISSUE TITLE
Backpropagate only on answer tokens during instruct fine-tuning

### ISSUE THREAD

**Issue by hanyin88** [facebookresearch/llama-recipes#294]
### 🚀 The feature, motivation and pitch

In the LLaMA-2 paper, it was mentioned that during instruct fine-tuning "We utilize an autoregressive objective and zero-out the loss on tokens from the user prompt, so as a result, we backpropagate only on answer tokens."

I am wondering has this function being implemented in the current recipe?  This seems  to be a highly valuable idea especially when doing fine tuning for a summarization task, where the prompt could be very long.

**Comment by HamidShojanazeri** [comment]
@hanyin88 yes, we mask out the prompt in our datasets and only backpropagate only on answer tokens.
